File: 02075. Decode the Slanted Ciphertext/solution.cs
/**
 * Problem: 2075. Decode the Slanted Ciphertext
 *
 * Difficulty: Medium
 *
 * Language: C#
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

public class Solution
{
  public string DecodeCiphertext(string encodedText, int rows)
  {
    // If only one row, the text is already in original order
    if (rows == 1)
      return encodedText;

    // Get total length of encoded text
    int totalLength = encodedText.Length;
    // Calculate number of columns in the grid
    int cols = totalLength / rows;

    // Initialize indices: row, column, position in encoded text
    int row = 0,
      col = 0,
      position = 0;

    // Store decoded characters
    System.Text.StringBuilder decoded = new System.Text.StringBuilder();

    // Traverse the grid in diagonal order
    while (position < totalLength)
    {
      // Add character at current position to decoded string
      decoded.Append(encodedText[position]);
      // Move to next row
      row++;

      // If reached bottom row, move to next column and reset row
      if (row == rows)
      {
        row = 0;
        col++;
      }

      // Calculate next position in encoded text
      position = row * (cols + 1) + col;
    }

    // Remove trailing spaces from decoded string
    return decoded.ToString().TrimEnd(' ');
  }
}

File: 02075. Decode the Slanted Ciphertext/solution.go
/**
 * Problem: 2075. Decode the Slanted Ciphertext
 *
 * Difficulty: Medium
 *
 * Language: Golang
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

func decodeCiphertext(encodedText string, rows int) string {
	// If only one row, the text is already in original order
	if rows == 1 {
		return encodedText
	}

	// Get total length of encoded text
	totalLength := len(encodedText)
	// Calculate number of columns in the grid
	cols := totalLength / rows

	// Initialize indices: row, column, position in encoded text
	row, col, position := 0, 0, 0

	// Store decoded characters
	decoded := make([]byte, 0)

	// Traverse the grid in diagonal order
	for position < totalLength {
		// Add character at current position to decoded string
		decoded = append(decoded, encodedText[position])
		// Move to next row
		row++

		// If reached bottom row, move to next column and reset row
		if row == rows {
			row = 0
			col++
		}

		// Calculate next position in encoded text
		position = row*(cols+1) + col
	}

	// Convert byte slice to string
	result := string(decoded)
	// Remove all trailing spaces from the result string
	result = strings.TrimRight(result, " ")

	// Return the decoded text without trailing spaces
	return result
}
